lib/dal/src/change_set.rs

```diff
@@ -439,12 +453,16 @@ impl ChangeSet {
             return Err(ChangeSetError::DvuRootsNotEmpty(ctx.change_set_id()));
         }
 
-        // if the change set status isn't approved, we shouldn't go
-        // locking stuff
-        if change_set.status != ChangeSetStatus::Approved {
-            return Err(ChangeSetError::ChangeSetNotApprovedForApply(
-                change_set.status,
-            ));
+        // WARNING(nick): we should only skip this status check if using sdf's protected apply logic.
+        if !dangerous_skip_status_check {
+            // if the change set status isn't approved, we shouldn't go
+            // locking stuff
+            let change_set = ChangeSet::get_by_id(ctx, ctx.change_set_id()).await?;
+            if change_set.status != ChangeSetStatus::Approved {
+                return Err(ChangeSetError::ChangeSetNotApprovedForApply(
+                    change_set.status,
+                ));
+            }
```

Comment:
We are using approval requirements instead of a database row's column's based value. We now know if approvals have gone stale and we need to know if the requirements are met at an _**sdf**_ level because it has access to SpiceDB. This will appear later in the PR.